Rework Cinnamon components

Caverns4 opened the pull request:
Themeing now works with the new dialogue styles in Linux Mint 21 and above.

Caverns4 committed:
Rework Cinnamon components

Themeing now works with the new dialogue styles in Linux Mint 21 and above.

Elbullazul commented:
Looks like you put a lot of effort into this patch; it is very appreciated.

I found a few regressions when testing your patches:
1. the scrollbars are very thin. I would revert to the previous code for them, since they look thicker and also include the "handle" in the slider
2. the toggles were switched from checkboxes to radioboxes. I think a checkbox better illustrates the on/off status of a toggle than a radiobutton, so I'd remove the new `Switches (to be used in menus)` block
3. the panel labels for the _Embedded_ and _Royale_ variants now show black text, which is hard to read, and doesn't follow the original white text of both themes.

Otherwise, it looks good to me, and ready to be merged once these issues are fixed.

Caverns4 committed:
Fixing most of the issues

Caverns4 commented:
As far as I can tell now, all the issues are fixed, only thing I couldn't figure out is that on the normal Royale theme the clock text is body. Its supposed to be black, but not have the shadow.

Elbullazul commented:
For Royale, you may look into the `applet.js` section (line 1998), and change all the `color: white/#fff` lines to `color: black`. To remove the bold font, edit the `.panel-button` class and remove the `font-weight: bold` line.

A few other bugs have appeared:
- Text alignment varies if an applet is hovered or not (ex. multiline clock label). You could standardize text alignment across variants by adding `text-align: center;` to the `.applet-box` class
- the run dialog text being white. It should be fixed by changing lines 1089-1090 to
```css
  color: black;
  caret-color: black;
```
- some notification text is white on light yellow, so unreadable. You can test this yourself by sending a test notification with `notify-send "hello-world" "some body"`
- I don't know if it's intended, but icon spacing in the right panel has been reduced in all variants, except Zune and Royale Dark. If it's intentional, could you apply the changes to them as well? If not, revert padding changes to the other variants.

Caverns4 committed:
Fixing the inconsistencies(except Run Dialogue)

The start button now looks nearly perfect, and so does the system tray. Popup notifications no longer show up white.

Caverns4 committed:
Fix run dialogue box inconistencies across all variants.

Caverns4 committed:
Twzr on pop-up notification buttons should also be black on all themes.

Caverns4 committed:
Paolish up loose ends, make modal dialogs match theme!

I believe that's everything fixed.
